### DISCUSSION TITLE
CSS styles auf bestimmte Seiten oder Tabs einschränken

### DISCUSSION THREAD

**Discussion by Bernd9000** [Zefau/ioBroker.jarvis#1183] [Q&A] [answered]
Ich möchte eine für das Handy optimierte Seite erstellen.
Z.B. möchte ich die Zeilenhöhe ändern mit folgendem Code:

.q-item--dark {height: 80px;}

Wie muß ich das als css code angeben das diese Änderung nur auf einer bestimmten Seite oder TAB angewendet wird?

**Comment by ma-gu-16** [comment]
@Bernd9000 
Die einzelnen Widgets scheinen classen zu haben z.b.  jarvis-widget-2xxxxx-xxxx-x-xxxxxx  evtl. genügt dir das?  Mit mediaqueries kannst du ja das dann ins Custom CSS Feld einfügen. 

@Zefau - es wäre evtl. noch gut eigenes JS nachladen zu können ?

**Comment by Bernd9000** [comment]
Das mit den widget classen habe ich noch nicht hingekriegt, liegt bestimmt daran das ich was falsch mache.

Versucht habe ich verschiedene Varianten:
Z.B.
.jarvis-widget-xxxxx-xxxx-xxxx-xxxx-xxxx {
.q-item--dark {
height: 100px;
}
}

oder :
[widgets="xxxxx-xxxx-xxxx-xxxx-xxxx"] {
.q-item--dark {
height: 100px;
}
}

**Comment by mcuiobroker** [comment]
.jarvis-widget-xxxxx-xxxx-xxxx-xxxx-xxxx .q-item--dark {
height: 100px;
}
Kann auch andersherum sein?
.q-item--dark .jarvis-widget-xxxxx-xxxx-xxxx-xxxx-xxxx {
height: 100px;
}

[widgets="xxxxx-xxxx-xxxx-xxxx-xxxx"] .q-item--dark {
height: 100px;
}
Hier auch entsprechend.

**Comment by Bernd9000** [comment]
Ja ,habs auch gerade herausgefunden.Danke euch.
Aber gibt es das auch für eine übergeordnete Seite ? Eventuell ".jarvis-page" ?
Das Problem ist, das ich das selbe Widget in mehreren Layouts benutze und somit ändere ich das in mehreren TABs

**Comment by mcuiobroker** [comment]
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/styles/tabs

**Reply by ma-gu-16** [reply]
das ist ja nice, kannte ich noch nicht...

**Comment by Bernd9000** [comment]
Ja, das hatte ich auf deiner Seite schon gesehen, funktioniert aber bei mir nicht, warscheinlich wieder ein Anwendungsfehler :-)
Bei mir sehen die Adressen so aus : .......index.html#/0/haussteuerung0
Also hab ich es so angegeben, funktioniert aber leider  nicht:

[href="#/0/haussteuerung0"] .q-item--dark {
height: 100px;
}

**Comment by mcuiobroker** [comment]
Welche Version hast du?
[href="#/haussteuerung0"] .q-item--dark {
height: 100px;
}

**Reply by ma-gu-16** [reply]
klappt mit umlauten leider nicht. ich mach ein issue auf.

**Reply by mcuiobroker** [reply]
für küche muss man diese Schreibweise nutzen
![image](https://user-images.githubusercontent.com/84818934/142379557-0ceb1fbe-afd2-4c26-8b43-e7a8658f104a.png)

**Reply by ma-gu-16** [reply]
ja danke habs bereits ergänzt und geschlossen

**Reply by Bernd9000** [reply]
Habe aber in hausteurung kein ü drin

**Reply by mcuiobroker** [reply]
Klappt es denn jetzt oder willst du es nicht mehr nutzen?

**Reply by Bernd9000** [reply]
Ich mach es jetzt mit @media all wie weiter unten  angegeben.
Aber das Seitenabhängig zu machen könnte nochmal wichtig werden.
Werd das später nochmal probieren...

**Reply by Bernd9000** [reply]
Also, die "href" Geschichte funktioniert wohl nur für die TABs.
Ich kann damit keine anderen Elemente erreichen.

**Reply by mcuiobroker** [reply]
In der Doku sind fast alle Elemente drin. Du kannst ja mal über die Suche einsteigen?
Oder auf das Element klicken, welches du beeinflussen möchtest, rechte Maustaste und auf "Untersuchen" klicken -> die Klassen, die dort verwendet werden, werden angezeigt.

**Comment by Bernd9000** [comment]
v3.0.0-alpha.125

**Comment by mcuiobroker** [comment]
Welches Element möchtest du beeinflussen? Das Widget als Ganzes?

**Comment by Bernd9000** [comment]
Ich will eine eigene Seite nur für das Handy erstellen weil die Schalter mit den Fingern schwer zu treffen sind.
Die einzelenen Geräte sind zu dicht zusammen, ich treffe dann häufig das falsche Gerät.
Also wollte ich in allen Tabs bei allen Widgets die Zeilenhöhe und die Schiftgröße vergrößern.

Normale Ansicht:
![Bild](https://user-images.githubusercontent.com/77005685/142380330-b3170184-77f6-4854-88d1-97ccfdc156f6.jpg)

Ziel :
![Bild - Kopie](https://user-images.githubusercontent.com/77005685/142380363-43bc6ec2-44ce-49b2-b739-b31584619ad5.jpg)

**Reply by mcuiobroker** [reply]
Das ist dann mehr was für @media-all. Da kannst du dann für alle Elemente die Schriftgröße ändern.
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/styles/besondere-klassen/media-all

**Reply by mcuiobroker** [reply]
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/styles/besondere-klassen/media-all#beispiel-handy-mehr-platz-zwischen-den-zeilen

**Comment by Bernd9000** [comment]
Danke, damit funktioniert es bei mir. :-)

**Comment by mcuiobroker** [comment, marked as answer]
Falls du die Schriftgröße anpassen möchtest:
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/styles/besondere-klassen/media-all#schriftgroesse-anpassen

**Reply by Bernd9000** [reply]
Danke

**Reply by mcuiobroker** [reply]
Hier noch mit SwitchAction Vergrößerung
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/styles/besondere-klassen/media-all#switchaction-vergroessern

**Comment by Bernd9000** [comment]
Besser wäre allerdings wenn ich css styles für eine Seite angeben könnte.
Die "@media all" Methode funktioniert nicht wenn ich das Handy im Querformat halte.

**Reply by mcuiobroker** [reply]
Im Querformat ist natürlich die max-width größer, schätze mal 860? Dann musst du anstatt 500px ->860px eintragen.
Genaue Werte bekommst du hiermit: https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/bildschirmgroesse

**Comment by Bernd9000** [comment]
1080 X 2340
Da hab ich dann aber an einem PC einen Monitor, der eine kleinere Auflösung hat.
Folge ist das am PC Monitor die css styles auch greifen was ja nicht sein soll.

**Reply by mcuiobroker** [reply]
Welche Auflösung ist denn das? 1080 x 2340? PC?

**Reply by Bernd9000** [reply]
Nein, mein Handy

**Reply by mcuiobroker** [reply]
Das ist aber nicht die Auflösung die vom HTML angezeigt wird?
Hast du mal die Bildschirmgröße ermittelt mit dem HTML was da zur Verfügung steht?
Welche Auflösung hat denn Dein PC-Monitor?

**Reply by Bernd9000** [reply]
Stimmt, hab ich vergessen, Browser setzen die Auflösung ja runter.
Ist 396 X 858
Sollte also auch damit funktionieren

**Reply by mcuiobroker** [reply]
Das ist die Auflösung im Hochformat, wie ist die im Querformat?
Es ist eine andere als 858 x 396

**Comment by Bernd9000** [comment]
![Screenshot_2021-11-18-23-09-23-791_org mozilla firefox](https://user-images.githubusercontent.com/77005685/142505092-5ec3cc04-aaf7-48df-b88f-7bf6da8b676f.jpg)

**Reply by mcuiobroker** [reply]
Also musst du statt 500px -> 805px eintragen.

**Reply by Bernd9000** [reply]
Jo, suche aber jetzt nach css Inhalts/Werte Selektoren um das doch noch auf bestimmte Seiten zu selektieren.

**Reply by mcuiobroker** [reply]
Kannst du das ein wenig mit Bildern veranschaulichen.
Was möchtest du auf bestimmten Seiten verändern?

**Reply by mcuiobroker** [reply]
Du hast Recht. Es fehlt ein Selektor für die Seite.
![image](https://user-images.githubusercontent.com/84818934/142511992-4fdb1190-e50d-427c-8905-418a084e90ec.png)

Bei jarvis-page sollte es eine id geben, mit der man ein Selektor für die Seite nutzen kann.
@zefau kannst du bitte einen entsprechenden Selektor für die Seite hinzufügen. Danke.

**Reply by Bernd9000** [reply]
> 
> 
> Kannst du das ein wenig mit Bildern veranschaulichen. Was möchtest du auf bestimmten Seiten verändern?

Im Moment brauch ich das selektieren auf einer bestimmten Seite noch nicht da deine Lösung ja funktioniert.
Aber das können wir bestimmt noch brauchen, es macht den css code auch übersichtlicher wenn ich mit wenigen Zeilen gleich eine ganze Seite mit mehreren Tabs verändern kann.Und da der Schwerpunkt zum verändern des Design/Theme nun mal im css code liegt suche ich jetzt nach weiteren Möglichkeiten und probiere einiges aus.Deine Seite ist dafür schon mal eine super Anlaufstelle.

**Reply by Bernd9000** [reply]
@mcuiobroker 
P.S. Ich kann übrigens auf deiner Seite mit dem Firefox Browser kein Code kopieren.
Bekannt?

**Reply by mcuiobroker** [reply]
Falls du etwas nicht finden kannst oder etwas falsch ist oder du andere css nutzt, immer direkt ansprechen, dann kann ich es ergänzen oder ändern.

**Reply by mcuiobroker** [reply]
Danke für den Hinweis. Ich geb es an gitbook weiter.

**Comment by Bernd9000** [comment]
Zefau hat jetzt mit v3.0.0-beta.6 die ID eingefügt, somit kann man im css code ein Selektor für eine jarvis Seite angeben.

Beispiel:
.jarvis-page-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx .q-item__section--main {
	min-height: 80px;
}

Hier wird auf der gesamten jarvis Seite in allen Tabs und Widgets mit der ID xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx die Zeilenhöhe vergrößert.
